### PR TITLE
gui auth

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Code for auth features on Frontend

### CODE REVIEW

File: gui/src/context/taipyReducers.ts [hunk 1]
```diff
@@ -163,6 +170,7 @@ export const INITIAL_STATE: TaipyState = {
         : undefined,
     id: getLocalStorageValue("TaipyClientId", ""),
     menu: {},
+    token: localStorage && localStorage.getItem("token") || undefined,
```
Comment (FredLL-Avaiga) [inline gui/src/context/taipyReducers.ts]: Why not use getLocalStorageValue ?

File: taipy/gui/gui.py [hunk 2]
```diff
@@ -1033,7 +1033,17 @@ def _set_frame(self, frame: FrameType):
             raise RuntimeError("frame must be a FrameType where Gui can collect the local variables.")
         self.__frame = frame
 
-    def run(self, run_server: bool = True, run_in_thread: bool = False, ssl_context: t.Optional[t.Union[ssl.SSLContext, t.Tuple[str, t.Optional[str]], t.Literal['adhoc']]] = None, **kwargs) -> t.Optional[Flask]:
+    def _set_state(self, state: State):
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui.py:1080]: I suppose this is used by gui enterprise ?

**Review by FredLL-Avaiga — APPROVED** [review]
need to review enterprise

File: gui/src/context/taipyReducers.ts (hunk 1 above)
Comment (dinhlongviolin1) [reply gui/src/context/taipyReducers.ts]: You are right thanks!

File: taipy/gui/gui.py (hunk 2 above)
Comment (dinhlongviolin1) [reply taipy/gui/gui.py:1080]: Yes you are correct! The new state class is used by the enterprise

**Review by FabienLelaquais — APPROVED** [review]
👍

**Pull request merged**